crates/flox-rust-sdk/src/models/search.rs

```diff
@@ -288,23 +286,28 @@ pub fn do_search(search_params: &SearchParams) -> Result<(SearchResults, ExitSta
 
     debug!("running search command {:?}", pkgdb_command);
     let mut pkgdb_process = pkgdb_command.spawn().map_err(SearchError::PkgDbCall)?;
-
-    // SAFETY: Could panic if somehow we aren't capturing `stdout`, but
-    //         we _need_ to capture `stdout` to read the search results
-    //         anyway. This is an error you would absolutely encounter
-    //         during integration tests, so it's safe to unwrap here.
-    let output_reader = BufReader::new(pkgdb_process.stdout.take().unwrap());
-    let err_reader = BufReader::new(pkgdb_process.stderr.take().unwrap());
-
-    let results = SearchResults::collect_results(output_reader, err_reader)?;
-
+    let stdout = pkgdb_process.stdout.take();
+    if stdout.is_none() {
+        return Err(SearchError::PkgDbStdout);
+    }
+    let deserializer = serde_json::Deserializer::from_reader(stdout.unwrap());
+    let mut count = None;
+    let mut results = Vec::new();
+    for maybe_record in deserializer.into_iter() {
+        let record = maybe_record.map_err(SearchError::Deserialize)?;
+        debug!("record = {:?}", record);
+        match record {
+            Record::Error(err) => return Err(SearchError::PkgDb(err)),
```

Comment:
If we return here does RAII kill the pkgdb process on drop? Do we care and can there be more than one error result anyway?

Reply:
Because we called `spawn` it won't kill the child process on drop. I'm not sure we care, but we should probably kill it just to be safe so it's not holding a database lock longer than necessary. There won't be more than one error on `stdout` that I'm aware of. If there is I'd say that's probably an armadillo thing.